src/main/java/com/ra4king/circuitsim/simulator/components/wiring/Clock.java

```diff
@@ -78,113 +129,120 @@ void startClock(int hertz) {
 					}
 				}
 			});
-			
+
 			clockThread.setName("Clock thread");
 			clockThread.setDaemon(true);
-			
+
 			currentClock = clockThread;
 			clockThread.start();
 		}
-		
+
 		void stopClock() {
 			if(currentClock != null) {
 				Thread clockThread = currentClock;
-				
+
 				currentClock.interrupt();
 				currentClock = null;
 				lastTickCount = 0;
-				
-				while(clockThread.isAlive()) {
-					Thread.yield();
-				}
+
+//				while(clockThread.isAlive()) {
```

Comment:
This was because of a bug that caused the application to crash if the debug is triggered on the edge of the clock. I resolved this instead by not yielding if the clock was stopped by the debug statement.